cmd/horologium/main.go

```diff
@@ -199,15 +203,23 @@ func sync(prowJobClient ctrlruntimeclient.Client, cfg *config.Config, cr cronCli
 			}
 			continue
 		}
-		if !shouldTrigger {
+		if !shouldTrigger && p.RetriggerFailed == nil {
 			logger.WithFields(logrus.Fields{
 				"previous-found": previousFound,
 				"name":           p.Name,
 				"job":            p.JobBase.Name,
 			}).Debug("Trigger time has not yet been reached.")
 		}
-		if !previousFound || shouldTrigger {
-			prowJob := pjutil.NewProwJob(pjutil.PeriodicSpec(p), p.Labels, p.Annotations,
+
+		if !previousFound || shouldTrigger || shouldTriggerFailedRun(j, p, now, logger, previousFound) {
+			labels := p.Labels
```

Comment:
Well this still modifies the config, `labels` is not a copy, just a reference to the original map.